### ISSUE TITLE
Add a music library tab to the main page

### ISSUE THREAD

**Issue by alex-esc** [TeamNewPipe/NewPipe#6158]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Now that SoundCloud and Bandcamp support has been added NewPipe could implement a music library to separate video content and audio / music content. Because you can add content to your playlist tab you can easily make a music playlist out of yt videos and songs on bandcamp but we currently can't separate music playlists from video playlist. 

This is especially confusing if you bookmark or add an entire album as a playlist because your video playlists and albums you'd saved are all mixed up.

A cool feature that would solve this is to add a separate section or tab on the main page for music and videos. This would need to be accompanied with an additional option on the "add to" button that allows users to add this video or song to ether your video playlists or to your music library. Your music library section would work just fine as the playlists tab currently does, expect it's independen from your video playlists and it's called "music library"

Here's an example of how I imagine the feature:

You can add any song (form youtube, soundlcoud, bandcamp) to your library with the add to button

![image](https://user-images.githubusercontent.com/35586515/115960852-48baeb00-a4d9-11eb-8363-40da3334c096.png)

you could also add a whole album playlist:

![image](https://user-images.githubusercontent.com/35586515/115961267-583b3380-a4db-11eb-9766-e8f80e4608b5.png)

But it adds this additional prompt:

![image](https://user-images.githubusercontent.com/35586515/115963583-8faedd80-a4e5-11eb-9b79-cbc1d2c7a45f.png)

And you can browse your albums and music playlists on a dedicated tab on the main page labeled Music library or music like in this picture:

![image](https://user-images.githubusercontent.com/35586515/115963646-e0263b00-a4e5-11eb-8517-8f4203b825e9.png)

Separating music playlist from video playlists would make NewPipe a viable alternative to music streaming services like Spotify and apple music. This would be very useful for music fans and FOSS enthusiasts because as far as I know there is no good FOSS streaming app or service that has access to such a big library of available mainstream music to stream. A big turnoff for going full FOSS is that not all apps have viable alternatives, and in my view this is the case for music streaming. It is true that underground alternatives for music streaming exist (like funkwhale) but your average user just wants to find a playlist of pop music or listen to an album or two from a major label. NewPipe can play a role here!

I think this would be a great fit for NewPipe because the app already has most of the features that streaming apps do. Search for music, queue songs to play next, listen on the background, download songs, make playlists and share URL's of individual tracks. Practically the only thing missing is a music library separate from a video playlist library.

Currently I have my favorite albums just added as playlists, kind of annoying to have cat video playlists and music mixed up but it works. An alternative (that I have not tried yet) would be to install multiple instances of NewPipe and use one instance only for videos and the other only for music, this way you can separate music from non-music. But again the average user would not go through such struggle.

If becoming a video and music app hybrid goes against the vision for NewPipe I guess a fork and rebrand would be needed, but as I have no app development experience I would need some guidance.

**Comment by mhmdanas** [comment]
#5654 would be more general but would also fulfill this use case.

**Comment by fabriiii** [comment]
I agree, I think this app would be perfect with a dedicated music management system! I used to keep favourite songs/albums saved in the app, but it's really dispersive as of now

**Comment by AthenaZiff** [comment]
Yes, NewPipe is missing a YouTube Music tab
